ckanext/restricted/action.py

```diff
@@ -114,16 +114,37 @@ def _restricted_resource_list_accessible_by_user(context, resource_list, package
     restricted_resources_list = []
     user_name = logic.restricted_get_username_from_context(context)
     user_obj = context.get('auth_user_obj')
+    if user_obj:
+        user_obj.org_dict = _get_org_dict(user_name)
     for resource in resource_list:
         resource_dict = dict(resource)
         if not package_dict:
             package_dict = package_show(context, {'id': resource_dict['package_id']})
-        if logic.restricted_check_user_resource_access(user_name, resource_dict, package_dict, user_obj=user_obj).get('success', False):
+        auth_result = logic.restricted_check_user_resource_access(
+            user_name,
+            resource_dict,
+            package_dict,
+            user_obj=user_obj,
+            pkg_show=False
+        )
+        if auth_result.get('success', False):
             restricted_resources_list.append(resource_dict)
 
     return restricted_resources_list
 
 
+def _get_org_dict(user_name):
```

Comment:
Let's make it public function in the `logic` module in order to remove code duplication.